sink/datadog.go

```diff
@@ -293,33 +394,72 @@ func (s *Datadog) Send(ctx context.Context, m *blip.Metrics) error {
 		return nil
 	}
 
-	// send metrics via API
-	ddCtx := context.WithValue(
-		ctx,
-		datadog.ContextAPIKeys,
-		map[string]datadog.APIKey{
-			"apiKeyAuth": {
-				Key: s.apiKeyAuth,
-			},
-			"appKeyAuth": {
-				Key: s.apiKeyAuth,
-			},
-		},
-	)
+	if n-rangeStart > 0 {
+		if err := s.sendApi(ddCtx, dp[rangeStart:n]); err != nil {
+			apiErrors = append(apiErrors, err.Error())
+		}
+	}
 
-	payload := datadogV2.MetricPayload{
-		Series: dp,
+	if len(apiErrors) > 0 {
+		return fmt.Errorf("%s", strings.Join(apiErrors, "\n"))
 	}
 
-	// Send metrics to Datadog. The Datadog client handles everything; we just pass
-	// it data points.
-	_, r, err := s.metricsApi.SubmitMetrics(ddCtx, payload, *datadogV2.NewSubmitMetricsOptionalParameters())
+	return nil
+}
+
+// Update our estimated partition
+func (s *Datadog) updatePartitionSize(metric datadogV2.MetricSeries, currentMessageLength int) error {
+	size, err := s.estimateSize(metric)
 	if err != nil {
-		blip.Debug("error sending data points to Datadog: %s", err)
-		blip.Debug("error sending data points to Datadog: Http Response - %s", r)
+		return err
 	}
 
-	return err
+	// Figure out how many metrics can fit inside the max payload, but pad it slightly to control for headers, etc.
+	estPartitionSize := (MAX_PAYLOAD_SIZE - 1000) / size
```

Comment:
Here's another example of where partition size is confusing: comment says "how many _metrics_", but it's calculating a value in bytes (based off MAX_PAYLOAD_SIZE and current payload size). I don't see where code does something like "N metrics = B bytes, so new partition size is something fraction of N".